libraries/redshop/helper/product.php

```diff
@@ -5133,5 +5136,39 @@ public static function checkCompareProduct($productId)
         return '';
     }
 
+    /**
+     * @param   integer    $productId     Product ID
+     * @param   float    $productPrice  Product price
+     *
+     * @return  float
+     *
+     */
+    public function calculateProductApplyMassDiscount($productId, $productPrice)
+    {
+        $productId = (int)$productId;
+
+        if (!$productId) {
+            return '';
+        }
+        $product = Redshop\Product\Product::getProductById($productId);
+
+        $db    = JFactory::getDbo();
```

Comment:
@hoangvan149dhv From now on that help for easy to change to redSHOP API, code should using prefix `J`, It's must using full namespace instead: ex:` \Joomla\CMS\Factory:: ...`